libsmtutil/CHCSmtLib2Interface.cpp

```diff
@@ -183,12 +195,41 @@ std::string CHCSmtLib2Interface::querySolver(std::string const& _input)
 	if (m_queryResponses.count(inputHash))
 		return m_queryResponses.at(inputHash);
 
-	smtAssert(m_enabledSolvers.smtlib2 || m_enabledSolvers.eld);
-	if (m_smtCallback)
+	smtAssert(m_enabledSolvers.eld || m_enabledSolvers.z3);
+	smtAssert(m_smtCallback, "Callback must be set!");
```

Comment:
I don't think instance of this class makes sense without smt callback. How about we put this assert in constructor?

The same with requiring at least 1 solver to be enabled.